internal/clients/http/image_builder/image_client.go

```diff
@@ -44,66 +45,63 @@ func newImageBuilderClient(ctx context.Context) (clients.ImageBuilder, error) {
 }
 
 func (c *ImageBuilderClient) Ready(ctx context.Context) error {
+	logger := ctxval.Logger(ctx)
+
 	resp, err := c.client.GetReadiness(ctx, headers.AddImageBuilderIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Error().Err(err).Msgf("Readiness request failed for image builder: %s", err.Error())
+		logger.Error().Err(err).Msgf("Readiness request failed for image builder: %s", err.Error())
 		return err
 	}
 	defer resp.Body.Close()
-	if !parsing.IsHTTPStatus2xx(resp.StatusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Readiness response from image builder: %d", resp.StatusCode)
-		return ClientErr
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode)
+	if err != nil {
+		return fmt.Errorf("ready call: %w", err)
 	}
 	return nil
 }
 
 func (c *ImageBuilderClient) GetAWSAmi(ctx context.Context, composeID string) (string, error) {
-	ctxval.Logger(ctx).Info().Msgf("Getting AMI of image %v", composeID)
+	logger := ctxval.Logger(ctx)
+	logger.Trace().Msgf("Getting AMI of image %v", composeID)
+
 	imageStatus, err := c.fetchImageStatus(ctx, composeID)
 	if err != nil {
 		return "", err
 	}
-	ctxval.Logger(ctx).Debug().Msgf("Verifying AWS type")
+
+	logger.Trace().Msgf("Verifying AWS type")
 	if imageStatus.Type != UploadTypesAws {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Image is not AWS type")
-		return "", BadImageTypeErr
+		return "", fmt.Errorf("%w: expected image type AWS", UnknownImageTypeErr)
 	}
-	awsStatus, ok := imageStatus.Options.(map[string]interface{})
+	awsStatus, ok := imageStatus.Options.(map[string]interface{})["ami"]
 	if !ok {
-		return "", BadImageTypeErr
+		return "", AwsStatusNotFound
 	}
-	return awsStatus["ami"].(string), nil
+	return awsStatus.(string), nil
 }
 
 func (c *ImageBuilderClient) fetchImageStatus(ctx context.Context, composeID string) (*UploadStatus, error) {
-	ctxval.Logger(ctx).Info().Msgf("Fetching image status %v", composeID)
+	logger := ctxval.Logger(ctx)
+	logger.Trace().Msgf("Fetching image status %v", composeID)
+
 	resp, err := c.client.GetComposeStatusWithResponse(ctx, composeID, headers.AddImageBuilderIdentityHeader)
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Failed to fetch image status from image builder")
+		logger.Warn().Err(err).Msg("Failed to fetch image status from image builder")
 		return nil, fmt.Errorf("cannot get compose status: %w", err)
 	}
-	statusCode := resp.StatusCode()
-	if parsing.IsHTTPNotFound(statusCode) {
-		return nil, ComposeNotFoundErr
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Image builder replied with unexpected status while fetching image status: %v", statusCode)
-		return nil, ClientErr
-	}
 
-	err = verifyImage(ctx, resp.JSON200)
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
 	if err != nil {
-		ctxval.Logger(ctx).Warn().Err(err).Msg("Image status in not ready")
-		return nil, err
+		if errors.Is(err, clients.NotFoundError) {
+			return nil, ComposeNotFoundErr
```

Comment:
here again, is it wise to return internal errors from the http/xxx packages? How would I compare the error without importing the package? I want to compare with `clients` errors, because that's what I have imported, right?

Reply:
The idea is that all client functions may return either common errors or specific errors. The caller does not care as it should be using `error.Is/As` to extract errors.

Reply:
To elaborate, it is okay (IMHO and for the liter) to return an error that is an exported error from other package. What is not okay (and linter complains about) is when you get a generic `error` and return it unwrapped.

Reply:
but you loose the information about this being clients.NotFoundError completely, right? I'm not able to figure out that this was `clients.NotFoundError` by any use of `error.Is/As` or am I?

Reply:
Added wrapping to all errors in the last commit.